client/src/app/features/statistic/statistic.component.ts

```diff
@@ -0,0 +1,115 @@
+
+import { Content } from '../search/content.model';
+import { DateTime } from './date-time.model';
+import { StatisticService } from './statistic.service';
+
+@Component({
+  selector: 'lvzviz-statistic',
+  templateUrl: './statistic.component.html',
+  styleUrls: ['./statistic.component.css'],
+})
+export class StatisticComponent implements OnInit, AfterViewInit {
+
+  private map!: L.Map;
+  private heat: any;
+  dataLoaded = false;
+
+  minValue = 0;
+  maxValue = 0;
+  options!: Options;
+
+  constructor(private statisticService: StatisticService) { }
+
+  ngOnInit(): void {
+    const minmaxdate$ = this.statisticService.fetchDates('minmaxdate');
+    const last7days$ = this.statisticService.fetchDates('last7days');
+
+    forkJoin([minmaxdate$, last7days$]).subscribe((dates) => {
+      const minStepValue = this.determineDateValue(dates[0][0]).getTime();
+      const maxStepValue = this.determineDateValue(dates[0][1]).getTime();
+      const minDateValue = this.determineDateValue(dates[1][0]);
+      const maxDateValue = this.determineDateValue(dates[1][1]);
+      this.minValue = minDateValue.getTime();
+      this.maxValue = maxDateValue.getTime();
+      this.options = {
+        draggableRange: true,
+        translate: (value: number) => new Date(value).toDateString(),
+        stepsArray: this.generateSliderSteps(minStepValue, maxStepValue).map((date: number) => ({ value: date })),
+      };
+      /** workaround:
+       * The range slider needs steps on building but in this case they are not known before DOM rendering.
+       * They need to be loaded first, see above.
+       * After this the slider can be displayed.
+       */
+      this.dataLoaded = true;
+      this.statisticService.fetch(minDateValue.toISOString(), maxDateValue.toISOString()).pipe(
+        map(data => data.content),
+      ).subscribe(data => this.addToMap(data));
+    });
+  }
+
+  ngAfterViewInit(): void {
+    this.initMap();
+  }
+
+  /**
+   * Initialize map.
+   */
+  initMap(): void {
+    /** workaround:
+     * Images not loaded correctly so images are copied from node_modules leaflet folder into assets folder.
+     */
+    L.Icon.Default.imagePath = "assets/leaflet/";
+
+    this.map = L.map('map-statistic').setView([51.339695, 12.373075], 11);
+
+    L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+      attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+    }).addTo(this.map);
+
+    this.heat = (L as any).heatLayer([], {
+      radius: 25,
+      minOpacity: 0.5
+    });
+  }
+
+  private addToMap(content: Content[]): void {
+    var latlng: any[] = [];
+    content.forEach((c) => {
+      if (c.coords) {
+        latlng.push([c.coords.lat, c.coords.lon]);
+      }
+    });
+    this.heat.setLatLngs(latlng);
+    // TODO: what does this line do?
+    // currently now functional b/c _size is not defined
+    //if (this.map._size.y === 0) { } else {
```

Comment:
I am not sure about this line. Currently I commented it out b/c this gives "_size is undefined".
What does this line do?
Has this anything to do with the coords of the points?
There are currently differences.
![heatmap](https://user-images.githubusercontent.com/10551931/146269599-c4284226-702d-45eb-b3ec-74a810c3d472.png)

Reply:
I changed the code to
```
if (this.map.getSize().y !== 0) {
      this.map.addLayer(this.heat);
    }
```
But I don't know what this check does.

For the heatmap:
I found a reason why the maps are different. The results are different, on localhost the most articles are snippets only. The server seems to load and get the data in a loop so most articles are fulltext and so more words matches.